src/diffusers/loaders/unet.py

```diff
@@ -364,8 +377,59 @@ def _process_lora(
             if is_peft_version(">=", "0.13.1"):
                 peft_kwargs["low_cpu_mem_usage"] = low_cpu_mem_usage
 
-            inject_adapter_in_model(lora_config, self, adapter_name=adapter_name, **peft_kwargs)
-            incompatible_keys = set_peft_model_state_dict(self, state_dict, adapter_name, **peft_kwargs)
+            if hotswap:
+                try:
+                    from peft.utils.hotswap import check_hotswap_configs_compatible, hotswap_adapter_from_state_dict
+                except ImportError as exc:
+                    msg = (
+                        "Hotswapping requires PEFT > v0.14. Please upgrade PEFT to a higher version or install it "
+                        "from source."
+                    )
+                    raise ImportError(msg) from exc
```

Comment:
Can we check the version instead of relying on exception? We have `is_peft_version`

https://github.com/huggingface/diffusers/blob/97abdd2210a540c2e71aee63c80a22723031cd57/src/diffusers/utils/import_utils.py#L808

Reply:
Same comment as above.